command/agent/consul/service_client.go

```diff
@@ -1130,8 +1118,7 @@ func (c *ServiceClient) UpdateWorkload(old, newWorkload *WorkloadServices) error
 
 	// Start watching checks. Done after service registrations are built
 	// since an error building them could leak watches.
-	for _, service := range newIDs {
-		serviceID := MakeAllocServiceID(newWorkload.AllocID, newWorkload.Name(), service)
+	for serviceID, service := range newIDs {
```

Comment:
looks like nothing async happens (unless hash does something) with serviceID but I'm always a little worried about this gotcha instead of allocating a new string

Reply:
Entries are deleted from the map but nothing is updated.